internal/setup/setup.go

```diff
@@ -344,21 +353,32 @@ type yamlArchive struct {
 
 type yamlPackage struct {
 	Name      string               `yaml:"package"`
-	Archive   string               `yaml:"archive"`
-	Essential []string             `yaml:"essential"`
-	Slices    map[string]yamlSlice `yaml:"slices"`
+	Archive   string               `yaml:"archive,omitempty"`
+	Essential []string             `yaml:"essential,omitempty"`
+	Slices    map[string]yamlSlice `yaml:"slices,omitempty"`
 }
 
 type yamlPath struct {
-	Dir     bool    `yaml:"make"`
-	Mode    uint    `yaml:"mode"`
-	Copy    string  `yaml:"copy"`
-	Text    *string `yaml:"text"`
-	Symlink string  `yaml:"symlink"`
-	Mutable bool    `yaml:"mutable"`
+	Dir     bool    `yaml:"make,omitempty"`
+	Copy    string  `yaml:"copy,omitempty"`
+	Text    *string `yaml:"text,omitempty"`
+	Symlink string  `yaml:"symlink,omitempty"`
+	Mutable bool    `yaml:"mutable,omitempty"`
+
+	Until PathUntil `yaml:"until,omitempty"`
+	Arch  yamlArch  `yaml:"arch,omitempty"`
+	Mode  yamlMode  `yaml:"mode,omitempty"`
+}
 
-	Until PathUntil `yaml:"until"`
-	Arch  yamlArch  `yaml:"arch"`
+func (yp *yamlPath) MarshalYAML() (interface{}, error) {
+	type flowPath yamlPath
```

Comment:
You can use a pointer here to avoid copying the object:
```go
type flowPath *yamlPath
```

Reply:
Done.